shelf/routes/artifact.py

```diff
@@ -125,17 +126,17 @@ def delete_metadata_property(container, bucket_name, path, item):
 
 @artifact.route("/<bucket_name>/artifact/_search", methods=["POST"])
 @decorators.foundation
-def root_search(container, bucket_name):
-    data = container.request.get_json(silent=True, force=True)
+@decorators.validate_request("schemas/search-request-criteria.json", {})
+def root_search(container, bucket_name, data):
     response = search(container, data)
 
     return response
 
 
 @artifact.route("/<bucket_name>/artifact/<path:path>/_search", methods=["POST"])
 @decorators.foundation
-def path_search(container, bucket_name, path):
-    data = container.request.get_json(silent=True, force=True)
+@decorators.validate_request("schemas/search-request-criteria.json", {})
```

Comment:
Look in the search portal. We validate against the schema twice for search criteria now.